embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/network/logging/EmbraceNetworkLoggingService.kt

```diff
@@ -49,113 +45,40 @@ internal class EmbraceNetworkLoggingService(
-        url: String,
-        httpMethod: String,
-        startTime: Long,
-        endTime: Long,
-        errorType: String?,
-        errorMessage: String?,
-        traceId: String?,
-        w3cTraceparent: String?,
-        networkCaptureData: NetworkCaptureData?
-    ) {
-        val duration = max(endTime - startTime, 0)
-        val strippedUrl = stripUrl(url)
-        val validTraceId = getValidTraceId(traceId)
-        val networkCall = NetworkCallV2(
-            url = strippedUrl,
-            httpMethod = httpMethod,
-            startTime = startTime,
-            endTime = endTime,
-            duration = duration,
-            traceId = validTraceId,
-            w3cTraceparent = w3cTraceparent,
-            errorMessage = errorMessage,
-            errorType = errorType
-        )
-
-        if (networkCaptureData != null) {
+    private fun logNetworkCaptureData(networkRequest: EmbraceNetworkRequest) {
+        if (networkRequest.networkCaptureData != null) {
             networkCaptureService.logNetworkCapturedData(
-                url,
-                httpMethod,
-                NETWORK_ERROR_CODE,
-                startTime,
-                endTime,
-                networkCaptureData,
-                errorMessage
+                networkRequest.url, //TODO: This used the non-stripped URL, is that correct?
+                networkRequest.httpMethod,
+                networkRequest.responseCode ?: NETWORK_ERROR_CODE,
+                networkRequest.startTime,
+                networkRequest.endTime,
+                networkRequest.networkCaptureData,
+                networkRequest.errorMessage
             )
         }
-        processNetworkCall(callId, networkCall)
-    }
-
-    override fun cleanCollections() {
-        clearNetworkCalls()
-        // re-fetch limits in case they changed since they last time they were fetched
-        defaultPerDomainSuffixCallLimit = configService.networkBehavior.getNetworkCaptureLimit()
-        domainSuffixCallLimits = configService.networkBehavior.getNetworkCallLimitsPerDomainSuffix()
     }
 
     /**
-     * Process network calls to be ready when the session requests them.
+     * Process network calls to assert that no limits are exceeded.
      *
-     * @param callId      the unique ID that identifies the specific network call instance being recorded
-     * @param networkCall that is going to be captured
+     * @param networkRequest the network request to process
      */
-    private fun processNetworkCall(callId: String, networkCall: NetworkCallV2) {
+    private fun processNetworkRequest(networkRequest: EmbraceNetworkRequest) {
         // Get the domain, if it can be successfully parsed. If not, don't log this call.
-        val domain = networkCall.url?.let {
-            getDomain(it)
-        } ?: return
+        val domain = getDomain(
+            stripUrl(networkRequest.url)
+        ) ?: return
 
         synchronized(callsStorageLastUpdate) {
```

Comment:
we can remove this synchronization and only block if we are counting against the same domain